Scan/PS2/PS2Keyboard.c

```diff
@@ -0,0 +1,101 @@
+/*
+  PS2Keyboard.cpp - PS2Keyboard library
+  Copyright (c) 2007 Free Software Foundation.  All right reserved.
+  Written by Christian Weichel <info@32leaves.net>
+
+  ** Mostly rewritten Paul Stoffregen <paul@pjrc.com> 2010, 2011
+  ** Modified for use beginning with Arduino 13 by L. Abraham Smith, <n3bah@microcompdesign.com> * 
+  ** Modified for easy interrup pin assignement on method begin(datapin,irq_pin). Cuningan <cuninganreset@gmail.com> **
+
+  for more information you can read the original wiki in arduino.cc
+  at http://www.arduino.cc/playground/Main/PS2Keyboard
+  or http://www.pjrc.com/teensy/td_libs_PS2Keyboard.html
+
+  Version 2.4 (March 2013)
+  - Support Teensy 3.0, Arduino Due, Arduino Leonardo & other boards
+  - French keyboard layout, David Chochoi, tchoyyfr at yahoo dot fr
+
+  Version 2.3 (October 2011)
+  - Minor bugs fixed
+
+  Version 2.2 (August 2011)
+  - Support non-US keyboards - thanks to Rainer Bruch for a German keyboard :)
+
+  Version 2.1 (May 2011)
+  - timeout to recover from misaligned input
+  - compatibility with Arduino "new-extension" branch
+  - TODO: send function, proposed by Scott Penrose, scooterda at me dot com
+
+  Version 2.0 (June 2010)
+  - Buffering added, many scan codes can be captured without data loss
+    if your sketch is busy doing other work
+  - Shift keys supported, completely rewritten scan code to ascii
+  - Slow linear search replaced with fast indexed table lookups
+  - Support for Teensy, Arduino Mega, and Sanguino added
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include "PS2Keyboard.h"
+
+#define BUFFER_SIZE 45
+static volatile uint8_t buffer[BUFFER_SIZE];
```

Comment:
Mostly an FYI (as you may be able to reduce some code). I have a keystate storing API that I built for the ergodox interconnect (which isn't PS/2, but it still sends press/release instead of updating every cycle).

I had to use it again for the unit test infrastructure I'm implementing in the ktype branch (hopefully merging soon).
https://github.com/kiibohd/controller/blob/ktype/Scan/TestIn/scan_loop.c#L114

As the code is now, I think it's fine though.